kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -565,6 +565,14 @@ object StreamCoreExtensions:
             mapChunkParUnordered(Async.defaultConcurrency, defaultAsyncStreamBufferSize)(f)(using t1, t2, t3, i1, i2, ev, frame)
 
     end extension
+
+    extension [V, S, S2](stream: Stream[V, S] < S2)
+        /** Takes a Stream[V, S] in the context of S2 (i.e. Stream[V, S] < S2) and returns a Stream that fuses together both effect contexts
+          * S and S2 into a single Stream[V, S & S2].
+          */
+        inline def unwrap: Stream[V, S & S2] = Stream(stream.map(_.emit))
+    end extension
+
```

Comment:
This should probably be in kyo-prelude, since it is not restricted `IO` or `Async`. Also, can I recommend instead of making this an extension method, make it a method on the `Stream` companion object?:

```scala
object Stream:
   ...
   def unwrap[V, S, S2](stream: Stream[V, S] < S2): Stream[V, S & S2] = ???
```

This is more consistent with Kyo's core API. We can then add the extension method version in kyo-combinators.